thunder/tests/test_autocast.py

```diff
@@ -232,3 +232,27 @@ def foo(x, w, b=None):
 
         for eg, jg in zip(eager_grads, jit_grads):
             torch.testing.assert_close(eg, jg, rtol=5e-2, atol=5e-2)
+
+
+@pytest.mark.parametrize("requires_grad", [False, True])
+def test_autocast_torch_matmul(requires_grad):
+    def foo(a, b):
+        output = torch.matmul(a, b)
+        return output
+
+    a = torch.rand([3, 1, 2], dtype=torch.float, device="cpu", requires_grad=requires_grad)
```

Comment:
It might be interesting to test both a float x bfloat16 input, and a float x float input, on both cpu and cuda devices